src/search-manager/data/api.ts

```diff
@@ -159,19 +162,27 @@ export async function fetchSearchResults({
 
   const blockTypesFilterFormatted = blockTypesFilter?.length ? [blockTypesFilter.map(bt => `block_type = ${bt}`)] : [];
 
+  const problemTypesFilterFormatted = problemTypesFilter?.length ? [problemTypesFilter.map(pt => `content.problem_types = ${pt}`)] : [];
+
   const tagsFilterFormatted = formatTagsFilter(tagsFilter);
 
   const limit = 20; // How many results to retrieve per page.
 
+  // To filter normal block types and problem types as 'OR' query
+  const typeFilters = [[
+    ...blockTypesFilterFormatted,
+    ...problemTypesFilterFormatted,
+  ].flat()];
+
```

Comment:
https://github.com/user-attachments/assets/46819f44-0170-4130-ac8f-2a8afb3fd021

Because of the OR condition between block types and problem types, filtering problem types after selecting whole problem block does not work. We can probably remove problem block type if problem types are selected.

Reply:
Updated: https://github.com/openedx/frontend-app-course-authoring/pull/1207/commits/f744344c976c4b8d9df594a446d6ac2b15c74f2a